pkg/web/apps/apps_http_handler.go

```diff
@@ -45,28 +47,79 @@ func (a *httpHandler) GetApps(c echo.Context) error {
 	return c.JSON(http.StatusOK, apps)
 }
 
-// GetAppByID returns an app with the ID in JSON format from the AppService
-// TODO: Implement
 //GetAppById returns a app with the ID in JSON format from the AppService
 func (a *httpHandler) GetAppByID(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+	app, err := a.Service.GetAppByID(c.Param("id"))
 
 	if err != nil {
 		return err
 	}
 
-	return c.JSON(http.StatusOK, apps)
+	return c.JSON(http.StatusOK, app)
 }
 
-// UpdateApp returns an app updated with the ID in JSON format from the AppService
-// TODO: Implement
 //UpdateApp returns a app updated with the ID in JSON format from the AppService
-func (a *httpHandler) UpdateApp(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+func (a *httpHandler) UpdateAppVersions(c echo.Context) error {
+	// Validations
+
+	versions := []models.Version{}
+	errV := json.NewDecoder(c.Request().Body).Decode(&versions)
+
+	// check if the data sent is in the correct format
+	if errV != nil {
+		return httperrors.BadRequest(c, "Invalid data")
+	}
+
+	// Check if versions were sent all the body is empty
+	if len(versions) == 0 {
+		return httperrors.BadRequest(c, "No version(s) was sent.")
+	}
+
+
+	// FIXME: When the GetAppByID be implemented
+	// Check if the app ID is valid
+	//app, err := a.Service.GetAppByID(c.Param("id"))
+	//if err != nil {
+	//	return httperrors.GetHTTPResponseFromErr(c, err)
+	//}
+	////Check if all versions has the same app id
+	//for i:=0; i < len(versions) ; i++ {
+	//	if versions[i].AppID != app.AppID {
+	//		return httperrors.BadRequest(c, "Version sent is not associated with the application informed")
+	//	}
+	//}
+
+	errUpdate := a.Service.UpdateAppVersions(versions)
+
+	if errUpdate != nil {
+		return httperrors.GetHTTPResponseFromErr(c, errUpdate)
+	}
+
+	return c.JSON(http.StatusOK, "")
```

Comment:
`HTTP status code 200 OK for a successful PUT of an update to an existing resource. No response body needed.`
Is it make sense for you @davidffrench? Any objection here?

Reply:
No objection at all, sending a 200 with no body is perfect.